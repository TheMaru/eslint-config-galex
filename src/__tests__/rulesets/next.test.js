const { createNextJsRules } = require('../../rulesets/next');

describe('createNextJsRules', () => {
  test('matches snapshot without next', () => {
    const project = {
      react: {
        isNext: false,
      },
    };

    expect(createNextJsRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with next', () => {
    const project = {
      react: {
        isNext: true,
      },
    };

    expect(createNextJsRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = '@next/next/no-unwanted-polyfillio';
    const level = 'off';

    const project = {
      react: {
        isNext: true,
      },
      rules: {
        [rule]: level,
      },
    };

    const defaultRuleset = createNextJsRules({
      ...project,
      rules: {},
    });

    const result = createNextJsRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
