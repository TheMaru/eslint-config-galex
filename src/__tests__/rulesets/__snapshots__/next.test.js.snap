// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNextJsRules allows passing custom rules 1`] = `
Object {
  "@next/next/missing-preload": "warn",
  "@next/next/no-css-tags": "warn",
  "@next/next/no-html-link-for-pages": "warn",
  "@next/next/no-sync-scripts": "warn",
  "@next/next/no-unwanted-polyfillio": "off",
}
`;

exports[`createNextJsRules matches snapshot with next 1`] = `
Object {
  "@next/next/missing-preload": "warn",
  "@next/next/no-css-tags": "warn",
  "@next/next/no-html-link-for-pages": "warn",
  "@next/next/no-sync-scripts": "warn",
  "@next/next/no-unwanted-polyfillio": "warn",
}
`;

exports[`createNextJsRules matches snapshot without next 1`] = `null`;
